### PR TITLE
[Improve][doc] add broker config for Oauth2

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation & Modification
add configuration to configure brokers to authenticate clients

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `no-need-doc` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

**Review by liudezhi2098 — APPROVED** [review]

File: site2/docs/security-oauth2.md [hunk 1]
```diff
@@ -195,6 +195,22 @@ This example shows how to configure OAuth2 authentication in Node.js client.
 ```
 > Note: The support for OAuth2 authentication is only available in Node.js client 1.6.2 and later versions.
 
+## Broker configuration
+To configure brokers to authenticate clients, add the following parameters to broker.conf or standalone.conf:
+```properties
+# Configuration to enable authentication and authorization
+authenticationEnabled=true
+authorizationEnabled=true
```
Comment (nodece) [inline site2/docs/security-oauth2.md:203]: I think should be `authorizationEnabled=true` removed.

File: site2/docs/security-oauth2.md [hunk 2]
```diff
@@ -195,6 +195,21 @@ This example shows how to configure OAuth2 authentication in Node.js client.
 ```
 > Note: The support for OAuth2 authentication is only available in Node.js client 1.6.2 and later versions.
 
+## Broker configuration
+To configure brokers to authenticate clients, add the following parameters to broker.conf or standalone.conf:
+```properties
+# Configuration to enable authentication and authorization
```
Comment (nodece) [inline site2/docs/security-oauth2.md:201]: Update this description.

File: site2/docs/security-oauth2.md [hunk 3]
```diff
@@ -195,6 +195,21 @@ This example shows how to configure OAuth2 authentication in Node.js client.
...
+authenticationEnabled=true
+authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderToken
+tokenPublicKey=/path/to/publicKey
+superUserRoles=[my-super-user-1,my-super-user-2]
```
Comment (nodece) [inline site2/docs/security-oauth2.md:205]: This also should be removed, it is authorization config.

File: site2/docs/security-oauth2.md (hunk 1 above)
Comment (nodece) [reply site2/docs/security-oauth2.md]: You should remove any authorization config like `superUserRoles`, and update the description about authentication.

File: site2/docs/security-oauth2.md (hunk 3 above)
Comment (nodece) [inline site2/docs/security-oauth2.md:205]:
```suggestion

```

File: site2/docs/security-oauth2.md (hunk 2 above)
Comment (nodece) [inline site2/docs/security-oauth2.md:201]:
```suggestion
# Configuration to enable authentication
```

File: site2/docs/security-oauth2.md (hunk 1 above)
Comment (nodece) [reply site2/docs/security-oauth2.md]: @liangyepianzhou

**Comment by Anonymitaet** [comment]
@momo-jun could you please help review this PR? Thanks

**Review by nodece — APPROVED** [review]
LGTM

File: site2/docs/security-oauth2.md [hunk 4]
```diff
@@ -195,6 +195,20 @@ This example shows how to configure OAuth2 authentication in Node.js client.
 ```
 > Note: The support for OAuth2 authentication is only available in Node.js client 1.6.2 and later versions.
 
+## Broker configuration
+To configure brokers to authenticate clients, add the following parameters to broker.conf or standalone.conf:
```
Comment (momo-jun) [inline site2/docs/security-oauth2.md:199]:
```suggestion
To enable OAuth2 brokers authentication in brokers, add the following parameters to the `broker.conf` or `standalone.conf` file.
```

**Review by momo-jun — CHANGES_REQUESTED** [review]
Minor suggestions.

File: site2/docs/security-oauth2.md [hunk 5]
```diff
@@ -195,6 +195,20 @@ This example shows how to configure OAuth2 authentication in Node.js client.
 ```
 > Note: The support for OAuth2 authentication is only available in Node.js client 1.6.2 and later versions.
 
+## Broker configuration
+To enable OAuth2 brokers authentication in brokers, add the following parameters to the `broker.conf` or `standalone.conf` file.
```
Comment (momo-jun) [inline site2/docs/security-oauth2.md:199]: `brokers` can be removed between `OAuth2` and `authentication`?

Comment (liangyepianzhou) [reply site2/docs/security-oauth2.md:199]: yes, I think so, and this is a suggestion given by your. 
My fault!  I did not review it carefully.

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**